runtime-transaction/src/compute_budget_program_id_filter.rs

```diff
@@ -1,37 +1,152 @@
 // static account keys has max
 use {
     agave_transaction_view::static_account_keys_frame::MAX_STATIC_ACCOUNTS_PER_PACKET as FILTER_SIZE,
-    solana_builtins_default_costs::MAYBE_BUILTIN_KEY, solana_sdk::pubkey::Pubkey,
+    solana_builtins_default_costs::{is_builtin_program, MAYBE_BUILTIN_KEY},
+    solana_sdk::pubkey::Pubkey,
 };
 
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub(crate) enum ProgramKind {
+    NotBuiltin,
+    Builtin { is_compute_budget: bool },
+}
+
 pub(crate) struct ComputeBudgetProgramIdFilter {
     // array of slots for all possible static and sanitized program_id_index,
     // each slot indicates if a program_id_index has not been checked (eg, None),
-    // or already checked with result (eg, Some(result)) that can be reused.
-    flags: [Option<bool>; FILTER_SIZE as usize],
+    // or already checked with result (eg, Some(ProgramKind)) that can be reused.
+    program_kind: [Option<ProgramKind>; FILTER_SIZE as usize],
 }
 
 impl ComputeBudgetProgramIdFilter {
     pub(crate) fn new() -> Self {
         ComputeBudgetProgramIdFilter {
-            flags: [None; FILTER_SIZE as usize],
+            program_kind: [None; FILTER_SIZE as usize],
         }
     }
 
     #[inline]
     pub(crate) fn is_compute_budget_program(&mut self, index: usize, program_id: &Pubkey) -> bool {
```

Comment:
this implementation is awkward. Can't use `get_or_insert` here because if instruction is not compute-budget, then we don't enough information to update filter. Looking for concise way to do same thing.

Reply:
Probably best to use two different filters. First pass we can use the compute budget filter and the second pass will use the builtin program filter if needed

Reply:
yea, try to take advantage of already identified compute-budget ixs, that probably not worth the unsatisfied code.

Reply:
split them in 36a2346465f6f4b722eaec70882ad92f4a2ba709